addons/web/views/report_templates.xml

```diff
@@ -511,6 +511,9 @@
                 <div t-if="report_type == 'pdf'" class="text-muted">
                     Page: <span class="page"/> / <span class="topage"/>
                 </div>
+                <div t-if="report_type == 'pdf' and show_invoice_number_in_footer" class="text-muted">
+                    <span t-field="o.name"/>
+                </div>
```

Comment:
You're adding it to only one layout here :/ Shouldn't it be done for all of them ? 
Plus, to me, this shouldn't be done in web if it's named show_invoice_number_in_footer. Maybe rename it to something like display_name_in_footer ?